### PR TITLE
Feat/Allow hyphens in schema names.

### PR DESCRIPTION
**Pull request by echen805**
## Problem

SQL interprets a dash in the schema name as a subtraction because its not quoted.

https://monadinc.atlassian.net/browse/ENG-439

## Proposed changes

~Adding a `.replace('-','_')` to the right places~
Quote the schema names to support hyphens.

## Types of changes

What types of changes does your code introduce to PipelineWise?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [ ] Description above provides context of the change
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Unit tests for changes (not needed for documentation changes)
- [ ] CI checks pass with my changes
- [ ] Bumping version in `setup.py` is an individual PR and not mixed with feature or bugfix PRs
- [ ] Commit message/PR title starts with `[AP-NNNN]` (if applicable. AP-NNNN = JIRA ID)
- [ ] Branch name starts with `AP-NNN` (if applicable. AP-NNN = JIRA ID)
- [ ] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [ ] Relevant documentation is updated including usage instructions

### CODE REVIEW

**Review by Jo-Con-El — CHANGES_REQUESTED** [review]
It should instead quote the DB name.

File: target_postgres/db_sync.py [hunk 1]
```diff
@@ -329,7 +329,7 @@ def table_name(self, stream_name, is_temporary=False, without_schema=False):
         if without_schema:
             return f'"{pg_table_name.lower()}"'
 
-        return f'{self.schema_name}."{pg_table_name.lower()}"'
+        return f'"{self.schema_name}""."{pg_table_name.lower()}"'
```
Comment (Jo-Con-El) [inline target_postgres/db_sync.py]: You put two double quotes here.

**Review by Jo-Con-El — CHANGES_REQUESTED** [review]

File: target_postgres/db_sync.py [hunk 2]
```diff
@@ -509,22 +509,22 @@ def create_schema_if_not_exists(self, table_columns_cache=None):
...
+            query = "CREATE SCHEMA IF NOT EXISTS '{}'".format(schema_name)
             self.logger.info("Schema '%s' does not exist. Creating... %s", schema_name, query)
             self.query(query)
 
             self.grant_privilege(schema_name, self.grantees, self.grant_usage_on_schema)
 
     def get_tables(self):
         return self.query(
-            'SELECT table_name FROM information_schema.tables WHERE table_schema = %s',
+            'SELECT table_name FROM information_schema.tables WHERE table_schema = "%s"',
```
Comment (Jo-Con-El) [inline target_postgres/db_sync.py]: I don't think double-quotes will do the trick:
```
monad=# SELECT table_name FROM information_schema.tables WHERE table_schema = acme-inc;
ERROR:  column "acme" does not exist
LINE 1: ...ROM information_schema.tables WHERE table_schema = acme-inc;
                                                              ^
monad=# SELECT table_name FROM information_schema.tables WHERE table_schema = "acme-inc";
ERROR:  column "acme-inc" does not exist
LINE 1: ...OM information_schema.tables WHERE table_schema = "acme-inc"...
                                                             ^
monad=# SELECT table_name FROM information_schema.tables WHERE table_schema = 'acme-inc';
 table_name 
------------
(0 rows)
```

File: target_postgres/db_sync.py [hunk 3]
```diff
@@ -329,7 +329,7 @@ def table_name(self, stream_name, is_temporary=False, without_schema=False):
         if without_schema:
             return f'"{pg_table_name.lower()}"'
 
-        return f'{self.schema_name}."{pg_table_name.lower()}"'
+        return f'"{self.schema_name}"."{pg_table_name.lower()}"'
```
Comment (Jo-Con-El) [inline target_postgres/db_sync.py:332]: I'm fairly sure that this won't work.  Double quotes are used for identifiers, but single quotes are used for strings (which is what you want here).

**Review by Jo-Con-El — CHANGES_REQUESTED** [review]
There are a million other places where this is going to be a problem, I think:
Line 457:
```
query = "GRANT USAGE ON SCHEMA {} TO GROUP {}".format(schema_name, grantee)
```
Line 462:
```
query = "GRANT SELECT ON ALL TABLES IN SCHEMA {} TO GROUP {}".format(schema_name, grantee)
```

**Review by cnuss — APPROVED** [review]

**Pull request merged**